app/code/Codifi/Sales/view/adminhtml/templates/order/view/info.phtml

```diff
@@ -223,14 +223,20 @@ $allowedAddressHtmlTags = ['b', 'br', 'em', 'i', 'li', 'ol', 'p', 'strong', 'sub
         <?php endif; ?>
     </div>
 </section>
+<?php
+$hide = false;
+if (\Codifi\Sales\Helper\Config::ORDER_TYPE_LABEL_CREDIT_HOLD && $viewModel->isCreditHoldConfigEnabled()){
+    $hide = \Codifi\Sales\Helper\Config::ORDER_TYPE_LABEL_CREDIT_HOLD;
+}
+?>
 <script>
     require([
         'jquery',
         'jquery/ui'
     ], function($) {
         $(document).ready(function($) {
             var order_type = $('#order_type').text();
-            if (order_type === <?= \Codifi\Sales\Helper\Config::ORDER_TYPE_LABEL_CREDIT_HOLD ?>) {
+            if (order_type === "<?= $hide ?>") {
```

Comment:
As I see `$hide` is boolean and you change type of the variable to string. Also `order_type` is string and you try to compare it with boolean/string using `===` which means that value and type must be equal.
The idea is to check if config enabled before adding script because if no, it's no need to check type and hide button.
If config enabled, you need to check type. Also it's better to save value from php to js var and compare it `order_type`.